astropy/coordinates/baseframe.py

```diff
@@ -519,9 +519,7 @@ def _infer_data(self, args, copy, kwargs):
 
         if len(args) > 0:
             raise TypeError(
-                "{}.__init__ had {} remaining unhandled arguments".format(
-                    self.__class__.__name__, len(args)
-                )
+                f"{self.__class__.__name__}.__init__ had {len(args)} remaining unhandled arguments"
```

Comment:
This can be shortened slightly:
```suggestion
                f"{type(self).__name__}.__init__ had {len(args)} remaining unhandled arguments"
```

Reply:
@eerovaher , does that mean this is another one where someone needs to make extra changes from their fork? If so, I will close this PR without merge. Please advise. Thanks!

Reply:
Ruff is converting `str.format()` calls to f-strings in the simplest possible way, but I don't think that simplest is best in many cases. I will update the relevant code manually.

Reply:
Thanks. Appreciate it!